src/main/java/com/epam/container/TransportList.java

```diff
@@ -0,0 +1,328 @@
+package com.epam.container;
+
+import com.epam.transport.Transport;
+
+import java.util.*;
+import java.util.function.Predicate;
+
+/**
+ * Class TransportList that implements and contains methods of List interface.
+ */
+public class TransportList implements List<Transport> {
+
+    private final int NUMBER_OF_FIELDS = 4;
+    private final static int INITIAL_CAPACITY = 5;
+    private int size;
+    Transport[] array;
+
+    public TransportList() {
+        this(INITIAL_CAPACITY);
+    }
+
+    public TransportList(int capacity) {
+        this.size = 0;
+        array = new Transport[capacity];
+    }
+
+    /**
+     * Returns the number of elements in this list.
+     */
+    @Override
+    public int size() {
+        return size;
+    }
+
+    /**
+     * Returns true if this list contains no elements
+     */
+    @Override
+    public boolean isEmpty() {
+        return size == 0;
+    }
+
+    /**
+     * @param o - element whose presence in this list is to be tested
+     * @return true if this list contains the specified element
+     */
+    @Override
+    public boolean contains(Object o) {
+        return indexOf(o) != -1;
+    }
+
+    /**
+     * Returns an iterator over the elements in this list in proper sequence.
+     */
+    @Override
+    public Iterator<Transport> iterator() {
+        return new Iterator<Transport>() {
+            int counter = 0;
+            @Override
+            public boolean hasNext() {
+                return counter < size;
+            }
+
+            @Override
+            public Transport next() {
+                if(!hasNext()) {
+                    throw new NoSuchElementException();
+                }
+                return array[counter++];
+            }
+        };
+    }
+
+    /**
+     * Returns an iterator over the elements in specified by Predicate sequence
+     * @param predicate - a condition that will be used to determine in what sequence to iterate
+     * @return Iterator
+     * @param <T> - A type
+     */
+    public <T> Iterator<T> iterator(Predicate<T> predicate) {
+        return new TransportListIterator<>(this, predicate);
```

Comment:
u won't use this TransportListIterator somewhere else, so, u r free to use inner class